arch/x86/x86.c

```diff
@@ -73,8 +94,36 @@ void Z_INTERNAL x86_check_features(void) {
         x86_cpu_has_tzcnt = ebx & 0x8;
         // check AVX2 bit
         x86_cpu_has_avx2 = ebx & 0x20;
+        x86_cpu_has_avx512 = ebx & 0x00010000;
+        x86_cpu_has_avx512vnni = ecx & 0x800;
     } else {
         x86_cpu_has_tzcnt = 0;
         x86_cpu_has_avx2 = 0;
     }
+
+
+    if (intel_cpu) {
+        /* All of the Knights Landing and Knights Ferry _likely_ benefit
+         * from the AVX512 adler checksum implementation */
+        if (family == 0xb) {
+            x86_cpu_well_suited_avx512 = 1;
+        } else if (family == 0x6) {
+            if (model == 0x5 && extended_model == 0x5) {
+                /* Experimentally, on skylake-x and cascadelake-x, it has been
+                 * unwaiveringly faster to use avx512 and avx512 vnni */
+                x86_cpu_well_suited_avx512 = 1;
+            } else if (model == 0xa && extended_model == 0x6) {
+                /* Icelake server */
+                x86_cpu_well_suited_avx512 = 1; 
+            } else if (model == 0xf && extended_model == 0x8) {
+                /* Saphire rapids */
+                x86_cpu_well_suited_avx512 = 1;
+            }
+
+            /* Still need to check whether Rocket Lake and/or AlderLake
```

Comment:
Alder Lake lost its AVX512 privileges in the recent microcode update.

Reply:
I had read that they did that as well.

Reply:
Yes, though, some users may decide to unpatch their microcode as things like RPCS3 benefits from it.  In any case, if it is supported, I believe cpuid will say so, so we won't inadvertently enable illegal instructions or anything.

I am curious how an unpatched AlderLake does here.